jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -370,6 +370,11 @@ public void addEventListener(final EventListener listener)
     /* ------------------------------------------------------------ */
     private MultiMap<String> getParameters()
     {
+        if (_parameters != null)
```

Comment:
I think this test is correct, but in the wrong place.
The lifecycle of parameters is stupidly complex because of request dispatchers... hence the previous solution of restoring parameters etc.

Thus I think this test needs to be moved down to after the extraction tests for content and query parameters.
Also I think that those extraction methods should both null `_parameters` if they perform an extraction.